### ISSUE TITLE
no-import-test-files incorrectly flags modules that end in .js

### ISSUE THREAD

**Issue by mearns** [avajs/eslint-plugin-ava#215]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/215/badge.svg" />](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/215)

Using all the default options, the following line in a test file is incorrectly flagged as violating the no-import-test-files rule:

```javascript
var highlight = require('highlight.js')
```

It appears that any module name ending in `.js` will similarly be flagged as a violation. However, the semantics for node's `require` function will _only_ look for local files if they start with a dot, so this will always be importing a module, not a test file.

> There is a $30.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/215).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $30.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/49722492/issues/215) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/49722492/issues/215).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/users/44827199/repositories) on IssueHunt to raise funds.

**Comment by GMartigny** [comment]
Hi @mearns,

I cannot reproduce your issue. Could you tell which version of `eslint` and `eslint-plugin-ava` you're using ? Could you maybe provide a simple repo showing the configuration you are using ?

Thanks

**Comment by mearns** [comment]
Sure, you can clone this repo and checkout the "eslint-plugin-ava-215" branch: https://github.com/mearns/lowlight/tree/eslint-plugin-ava-215, then run "npm run format". It runs `xo` which runs eslint with the ava/no-import-test-files rule, and gives the following violation:

```
  test/index.js:3:17
  ✖  3:17  Test files should not be imported  ava/no-import-test-files

  1 error
```
Note that this is with eslint-plugin-ava version 5.1.1

**Comment by IssueHuntBot** [comment]
@sindresorhus has rewarded $27.00 to @gmartigny. [See it on IssueHunt](https://issuehunt.io/repos/49722492/issues/215)

- :moneybag: Total deposit: $30.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $3.00
